packages/contracts-sdk/src/lib/contracts-sdk.ts

```diff
@@ -357,137 +358,225 @@ export class LitContracts {
+    const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+
+    if (network === 'cayenne') {
+      return new ethers.Contract(StakingData.address, StakingData.abi, provider);
+    } else if (network === 'internalDev') {
+      const INTERNAL_DEV_API =
+        'https://lit-general-worker.getlit.dev/internal-dev-contract-addresses';
+
+      let data;
+      try {
+        // Fetch and parse the JSON data in one step
+        data = await fetch(INTERNAL_DEV_API).then((res) => res.json());
+      } catch (e: any) {
+        throw new Error(`Error fetching data from ${INTERNAL_DEV_API}: ${e.toString()}`);
+      }
+      // Destructure the data for easier access
+      const { config, data: contractData } = data;
+      const stakingContract = contractData.find((item: { name: string }) => item.name === 'Staking')
+        .contracts[0];
+      const { address_hash: address, ABI: abi } = stakingContract;
+
+      // Validate the required data
+      if (!config || !address || !abi) {
+        throw new Error('❌ Required contract data is missing');
+      }
+
+      return new ethers.Contract(address, abi, provider);
+    } else {
+      throw new Error(`Invalid network. Only cayenne and internalDev are supported.`);
+    }
+  }
+
+  public static getMinNodeCount = async (network: LitNetwork.Cayenne | LitNetwork.InternalDev) => {
+
+    const contract = await LitContracts.getStakingContract(network);
+
+    const configs = await contract['config']();
+    const minNodeCount = configs.minimumValidatorCount.toString();
+
+    if (!minNodeCount) {
+      throw new Error('❌ Minimum validator count is not set');
+    }
+    return minNodeCount;
+  }
+
+  public static getValidators = async (network: LitNetwork.Cayenne | LitNetwork.InternalDev): Promise<string[]> => {
+
+    const contract = await LitContracts.getStakingContract(network);
+
+    // Fetch contract data
+    const [activeValidators, currentValidatorsCount, kickedValidators] =
+      await Promise.all([
+        contract['getValidatorsStructsInCurrentEpoch'](),
+        contract['currentValidatorCountForConsensus'](),
+        contract['getKickedValidators'](),
+      ]);
+
+    const validators = [];
+
+    // Check if active validator set meets the threshold
+    if (
+      activeValidators.length - kickedValidators.length >=
+      currentValidatorsCount
+    ) {
+      // Process each validator
+      for (const validator of activeValidators) {
+        validators.push(validator);
+      }
+    } else {
+      console.log('❌ Active validator set does not meet the threshold');
+    }
+
+    // remove kicked validators in active validators
+    const cleanedActiveValidators = activeValidators.filter(
+      (av: any) => !kickedValidators.some((kv: any) => kv.nodeAddress === av.nodeAddress)
+    );
+
+    const networks = cleanedActiveValidators.map((item: any) => {
+      return `http://${intToIP(item.ip)}:${item.port}`;
```

Comment:
i dont think we want to hardcode `http://` as prod nodes will run in `https` but im unsure how we will know as if we attempt to connect over http on any of the ports it will fail i think